### PR TITLE
[WIP][DON'T MERGE] Replace oTracking with n-third-party-code

### PR DESCRIPTION
**Pull request by adambraimbridge**

### CODE REVIEW

**Comment by adambraimbridge** [comment]
@wheresrhys : can I come over to PR this with you?

File: src/js-setup.js [hunk 1]
```diff
@@ -112,21 +109,8 @@ JsSetup.prototype.init = function (opts) {
...
-					layout: oGrid.getCurrentLayout(),
-					orientation: oViewport.getOrientation()
-				}
-			});
-
-			// The page url and referrer automatically populated by o-tracking
-			oTracking.page({});
-		}
+		// Third-party-code (includes tracking and advertising)
+		nThirdPartyCode.init(flags);
```
Comment (adambraimbridge) [inline src/js-setup.js]: From @wheresrhys : "You need to decide whether this code should run before the application code, during the app execution, or after the app's done." See `JsSetup.prototype.bootstrap` below.

**Comment by wheresrhys** [comment]
If window load has already fired then https://github.com/Financial-Times/n-third-party-code/blob/master/src/main.js#L9 will not run. So that case needs to be handled. Not sure of the best cross browser way to do this, but maybe look at https://github.com/Financial-Times/o-autoinit and copy code from there

I would also put https://github.com/Financial-Times/n-third-party-code/blob/master/src/main.js#L9 in a separate function which can be called somewhere around https://github.com/Financial-Times/next-js-setup/blob/master/src/js-setup.js#L152. The window.load can happen before the application code runs if main.js is the last resource to load, so this would make sure ads always initialise after the application.

**Comment by adambraimbridge** [comment]
Thanks for your help @wheresrhys — it's ready for you to look at again.

File: src/window-loaded.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
+"use strict";
+
+var hasFired = {};
+
+function trigger(type) {
+  if (type in hasFired) return;
+  hasFired[type] = true;
+  document.dispatchEvent(new Event(type));
```
Comment (wheresrhys) [inline src/window-loaded.js]: I think you can get rid of all the DOMContentLoaded stuff everywhere. Once it's simplified to only do 'load', this should then be something like `new Event('ftNextLoad')` and in n-third-party listen for `ftNextLoad`

In fact, most of the code here can be deleted. I didn't realise the weird edge case handling stuff was to do with DOMContentLoaded as opposed to load

**Comment by adambraimbridge** [comment]
Once the CI builds start passing again this will be ready for greenlighting

**Pull request merged**